.github/workflows/trigger-deployments.yml

```diff
@@ -0,0 +1,53 @@
+name: Trigger deployment on test then prod
+
+on:
+  push:
```

Comment:
As written, this will trigger on every push to ANY branch, right?

I know we don't really do much dev work in here, but we probably should update this to only happen on pushes to main to give us more control.

Reply:
The tags filter below it should mean it effectively only runs when a tag is pushed. Maybe there is a better way, though:
```
on:
  push:
    tags:
    - '*'
```

A quick search reveals [a SO post suggesting the same](https://stackoverflow.com/questions/61891328/trigger-github-action-only-on-new-tags).

This part of the code had been working fine [in the previous incarnation](https://github.com/PhilanthropyDataCommons/deploy/pull/107/files#diff-43b13329f8257d4e946cab27f88b61d3b8738929640b86496e7169dc8da9f871L5).

Reply:
In other words, in this repo, we don't want to auto-push anything that gets merged to main, only when a tag gets pushed. On a push to main in the `service` repo, this will cause a tag to get pushed to this repo. Maybe that design is up for review too but it's been working well enough for the past couple years.

Reply:
@bickelj could we have BOTH -- e.g. "the push has to have a tag AND be to main"?

I think what separates this from the old way is now we're talking about pushing to production so stricter scrutiny is useful.

Reply:
@slifty By restricting it to a tag push already, we decouple deployments from code changes. To push a tag is a special action. The tag references a commit much like a branch does (internally I'm not sure there's much of a difference). If the tag name does not conform to the scripts' expected syntax, then a deployment won't happen. Perhaps we need to protect against accidental tag pushes. I guess I can conceive of a tag referencing some branch somewhere instead of the main branch. But the main branch can change too, away from a previously-tagged commit. Are you saying we need a rule somewhere that "only on a tag that references a presently-in-the-main-branch commit this flow should run"? That makes sense but I'm not sure how to do that yet. I can look into it.

Reply:
Right now, the auto-tag has happened only on the main branch, see https://github.com/PhilanthropyDataCommons/deploy/blob/main/.github/workflows/update-service-image.yml - but to your point, perhaps that needs to be strictly limited to main as well.

Reply:
I used [action-contains-tag](https://github.com/rickstaa/action-contains-tag) as an additional protection for production deployments. So now production deployments require both the test deployment to succeed *and* the pushed tag to be verified as part of the main branch. I could imagine a scenario where perhaps we'd like to push a tag on a branch that we'd like deployed to the test environment but not production which is why I didn't add the same check on the deployment to test.